File: src/main/scala/com/example/scala/PessoaOperations.scala
package com.example.scala

import java.util.Random

import scala.collection.mutable

/**
 * @author Giovanni Silva 
 *         09/03/15.
 */
object PessoaOperations {
  val nomes: List[String] = List("Mestre", "Dengoso", "Atchim", "Feliz", "Dunga", "Soneca", "Zangado")
  val sobrenomes: List[String] = List("Schwarzenegger", "Van Dame", "Lee", "Chan", "Stallone", "Lundgren", "Li")

  val random = new Random()

  def gerarPessoaRandon(idadeMaxima: Int): Pessoa = {
    val nome: String = nomes(random.nextInt(nomes.size))
    val sobrenome: String = sobrenomes(random.nextInt(sobrenomes.size))
    val idade: Int = random.nextInt(idadeMaxima)
    return new Pessoa(nome, sobrenome, idade)
  }

  def gerarLista(tamanho: Int): mutable.ListBuffer[Pessoa] = {
    val pessoas: mutable.ListBuffer[Pessoa] = mutable.ListBuffer.empty
    for (i <- 0 until tamanho) {
      pessoas.append(gerarPessoaRandon(50))
    }
    return pessoas
  }

  def imprimirLista[T](lista: mutable.ListBuffer[T]) {
    for (i <- 0 until lista.size) {
      println(lista(i))
    }
  }

}

File: src/main/java/com/example/java/PessoaOperations.java
package com.example.java;

import java.util.*;

/**
 * @author Giovanni Silva
 *         09/03/15.
 */
public class PessoaOperations {

    private static final List<String> nomes = Arrays.asList("Mestre", "Dengoso", "Atchim", "Feliz", "Dunga", "Soneca", "Zangado");
    private static final List<String> sobrenomes = Arrays.asList("Schwarzenegger","Van Dame","Lee","Chan","Stallone","Lundgren","Li");

    private static Random random = new Random();

    private static Pessoa gerarPessoaRandon(int idadeMaxima){
        String nome = nomes.get(random.nextInt(nomes.size()));
        String sobrenome = sobrenomes.get(random.nextInt(sobrenomes.size()));
        int idade = random.nextInt(idadeMaxima);
        return new Pessoa(nome,sobrenome,idade);
    }

    public static List<Pessoa> gerarLista(int tamanho){
        List<Pessoa> pessoas = new ArrayList<>(tamanho);
        for(int i =0;i<tamanho;i++){
            pessoas.add(gerarPessoaRandon(50));
        }
        return Collections.unmodifiableList(pessoas);
    }

    public static <T> void imprimirLista(List<T> lista){
        for(int i=0;i<lista.size();i++){
            System.out.println(lista.get(i));
        }
    }
}
